lang/src/main/kotlin/org/partiql/lang/util/ExprValueFormatter.kt

```diff
@@ -42,7 +43,15 @@ class ConfigurableExprValueFormatter(private val config: Configuration) : ExprVa
                 ExprValueType.MISSING -> out.append(MISSING_STRING)
                 ExprValueType.NULL -> out.append(NULL_STRING)
                 ExprValueType.BOOL -> out.append(value.scalar.booleanValue().toString())
-                ExprValueType.INT, ExprValueType.DECIMAL -> out.append(value.scalar.numberValue().toString())
+                ExprValueType.INT -> out.append(value.scalar.numberValue().toString())
+                ExprValueType.DECIMAL -> {
+                    val decimalValue = value.scalar.numberValue() as? BigDecimal
+                    out.append(decimalValue.toString())
+                    // If this is a decimal with scale to be 0, we need to add a decimal point to differentiate it from an integer
+                    if (decimalValue != null && decimalValue.scale() == 0) {
+                        out.append(".")
+                    }
+                }
```

Comment:
Originally, a decimal with scale to be 0 is printed the same as an integer. e.g. a decimal with value `1.0` is printed as `1`. The change here makes it printed as `1.`, which is a decimal in PartiQL syntax.

Reply:
This change is related to this PR, since people normally write a decimal with scale of 0 as integer. e.g. if the result is `1.0`, they can probably write it as `1`. But this is not correct since the data type does not match. The change applied here prints the actual value as `1.`, instead of just `1`, which reminds people of the data type.